ludwig/datasets/mixins/kaggle.py

```diff
@@ -0,0 +1,40 @@
+import os
+from pathlib import Path
+from kaggle.api.kaggle_api_extended import KaggleApi
+
```

Comment:
from contextlib import contextmanager

@contextmanager
def update_env(**kwargs):
    prev_env = {}
    for k, v in kwargs.items():
        if v is not None:
            prev_env = os.environ.get(k)
            os.environ[k] = v
    try:
         yield
    finally:
        for k, v in prev_env.items():
            os.environ[k] = v
```